src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -515,6 +517,100 @@ protected function initializeBundles()
     }
 
     /**
+     * Returns an array of bundles to register, with dependencies, ordered.
+     *
+     * @uses appendDependenciesRecursively
+     *
+     * @return BundleInterface[] An array of bundle instances.
+     */
+    protected function registeredDependencies()
+    {
+        $bundles = $this->registerBundles();
+        $children = $rootBundles = array();
+        $hasDependencies = false;
+
+        // Build up bundles as a hash with FQN for basis to rebuild again in correct order given dependencies
```

Comment:
FQN as an expression already exists in PHP, other php libs, and in general: https://en.wikipedia.org/wiki/Fully_qualified_name